### PR TITLE
Port push subcommand to rust

### LINKED ISSUE #53: Port "git-dit-push" to rust
What the title says.

### PR DESCRIPTION
**Pull request by neithernut**
Resolves #53.

### CODE REVIEW

**Comment by neithernut** [comment]
Looks like we can't use refspec globs for pushing. We'll have to expand those.

File: src/main.rs [hunk 1]
```diff
@@ -279,15 +279,15 @@ fn push_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
         // push a specific list of issues
         issues.map(Oid::from_str).abort_on_err()
               .map(|issue| repo.get_issue_refs(issue)).abort_on_err()
-              .flat_map(|refs| refs).abort_on_err()
-              .filter_map(|issue_ref| remote.issue_refspec_from_ref(issue_ref))
+              .flat_map(git2::References::names).abort_on_err()
```
Comment (matthiasbeyer) [inline src/main.rs]: I would put the `abort_on_err()` on the next line. It is better for readability.

File: src/main.rs [hunk 2]
```diff
@@ -279,15 +279,15 @@ fn push_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
...
+              .map(String::from)
               .collect()
     } else {
         try_or_1!(repo.get_issue_hashes("refs"))
             .abort_on_err()
             .map(|issue| repo.get_issue_refs(issue)).abort_on_err()
-            .flat_map(|refs| refs).abort_on_err()
-            .filter_map(|issue_ref| remote.issue_refspec_from_ref(issue_ref))
+            .flat_map(git2::References::names).abort_on_err()
```
Comment (matthiasbeyer) [inline src/main.rs]: Same here.

File: src/main.rs [hunk 3]
```diff
@@ -268,6 +268,39 @@ fn new_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
...
+fn push_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    // note: "remote" is always present since it is a required parameter
+    let mut remote = try_or_1!(repo.find_remote(matches.value_of("remote").unwrap()));
+
+    // accumulate the refspecs to push
+    let refspecs : Vec<String> = if let Some(issues) = matches.values_of("issue") {
+        // push a specific list of issues
+        issues.map(Oid::from_str).abort_on_err()
+              .map(|issue| repo.get_issue_refs(issue)).abort_on_err()
+              .flat_map(git2::References::names).abort_on_err()
```
Comment (matthiasbeyer) [inline src/main.rs]: Pls put the `abort_on_err()` call on the next line. Its much better for readability.

File: src/main.rs [hunk 4]
```diff
@@ -268,6 +268,39 @@ fn new_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
...
+              .map(String::from)
+              .collect()
+    } else {
+        try_or_1!(repo.get_issue_hashes("refs"))
+            .abort_on_err()
+            .map(|issue| repo.get_issue_refs(issue)).abort_on_err()
+            .flat_map(git2::References::names).abort_on_err()
```
Comment (matthiasbeyer) [inline src/main.rs]: and here.

File: src/main.rs (hunk 1 above)
Comment (neithernut) [inline src/main.rs]: I think it's _less_ readable that way. I'll push a fixup anyways.

Comment (neithernut) [inline src/main.rs]: Have a look at it again.

**Pull request merged**